mmv1/third_party/terraform/fwprovider/data_source_provider_config_plugin_framework.go

```diff
@@ -0,0 +1,231 @@
+package fwprovider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-google/google/fwmodels"
+	"github.com/hashicorp/terraform-provider-google/google/fwresource"
+	"github.com/hashicorp/terraform-provider-google/google/fwtransport"
+)
+
+// Ensure the data source satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &GoogleProviderConfigPluginFrameworkDataSource{}
+	_ datasource.DataSourceWithConfigure = &GoogleProviderConfigPluginFrameworkDataSource{}
+	_ fwresource.LocationDescriber       = &GoogleProviderConfigPluginFrameworkModel{}
+)
+
+func NewGoogleProviderConfigPluginFrameworkDataSource() datasource.DataSource {
+	return &GoogleProviderConfigPluginFrameworkDataSource{}
+}
+
+type GoogleProviderConfigPluginFrameworkDataSource struct {
+	providerConfig *fwtransport.FrameworkProviderConfig
+}
+
+type GoogleProviderConfigPluginFrameworkModel struct {
+	// Currently this reflects the FrameworkProviderConfig struct and ProviderModel in google/fwmodels/provider_model.go
+	// which means it uses the plugin-framework type system where values can be explicitly Null or Unknown.
+	//
+	// As part of future muxing fixes/refactoring we'll change this struct to reflect structs used in the SDK code, and will move to
```

Comment:
So this is saying one of the anticipated fixes for existing pf/sdk muxing issues is to rewrite the model in `google/fwmodels/provider_model.go`, which means we will also need to change it here too ?

Reply:
The fixes for the muxing issues is to replace use of FrameworkProviderConfig with using the Config struct that is used in the SDK code. That means eventually the google_provider_config_plugin_framework data source will receive all the input data from a Config struct and that's implemented with the SDK type system (string, vs types.StringType). But before I do that I need to get these tests in place using the old struct.

Basically this comment is me saying I'm implementing this all one way, and will immediately switch it up as soon as all the acceptance tests are in place and make it easier for someone to review the muxing fixes.